src/main/java/teammates/logic/core/Emails.java

```diff
@@ -765,30 +191,27 @@ private void generateLogReport(MimeMessage message) throws MessagingException {
         }
     }
     
-    public MimeMessage sendErrorReport(HttpServletRequest req, Throwable error) {
-        MimeMessage email = null;
-        try {
-            String requestMethod = req.getMethod();
-            String requestUserAgent = req.getHeader("User-Agent");
-            String requestPath = req.getServletPath();
-            String requestUrl = req.getRequestURL().toString();
-            String requestParam = HttpRequestHelper.printRequestParameters(req);
-            email = generateSystemErrorEmail(error,
-                                            requestMethod,
-                                            requestUserAgent,
-                                            requestPath,
-                                            requestUrl,
-                                            requestParam,
-                                            Config.getAppVersion());
-            forceSendEmailThroughGaeWithoutLogging(email);
-            log.severe("Sent crash report: " + Emails.getEmailInfo(email));
-        } catch (Exception e) {
-            log.severe("Crash report failed to send. Detailed error stack trace: "
-                     + TeammatesException.toStringWithStackTrace(error));
-            logSevereForErrorInSendingItem("crash report", email, e);
-        }
+    /**
+     * Sends the given {@code errorReport}.
+     */
+    public void sendErrorReport(MimeMessage errorReport) throws MessagingException {
+        forceSendEmailThroughGaeWithoutLogging(errorReport);
+        log.info("Sent crash report: " + Emails.getEmailInfo(errorReport));
+    }
     
-        return email;
+    /**
+     * Reports that a system {@code error} has occurred and the {@code errorReport} that is
+     * supposed to report it has failed to sent.<br>
+     * This method can be used when the usual error report sending fails to make sure that
+     * no stack traces are lost in the process.
+     * @param error the original error to be reported in {@code errorReport}
+     * @param errorReport the report that fails to send
+     * @param e the exception which causes {@code errorReport} to fail to send
+     */
+    public void reportErrorWithBackupChannel(Throwable error, MimeMessage errorReport, Exception e) {
+        log.severe("Crash report failed to send. Detailed error stack trace: "
+                   + TeammatesException.toStringWithStackTrace(error));
+        logSevereForErrorInSendingItem("crash report", errorReport, e);
```

Comment:
Although this was in the old code too, this looks odd. Doesn't this do log.severe twice?